frontend/src/components/AddOrEditStep.vue

```diff
@@ -1,75 +1,104 @@
 <script setup lang="ts">
+import { computed, watch } from "vue";
+import { TRACK_FIELD_INPUT, TRACK_FIELD_TYPE } from "@/constants";
+
 import { useTrackStore } from "@/store/track";
 import { useStepStore } from "@/store/step";
-import { TRACK_FIELD_INPUT, TRACK_FIELD_TYPE } from "@/constants";
-import { ref } from "vue";
 
 const trackStore = useTrackStore();
 const stepStore = useStepStore();
 
-let showAddStepModal = ref(false);
+const props = defineProps({
+  show: {
+    type: Boolean,
+    default: false,
+  },
+  edit: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const emit = defineEmits(["close"]);
+
+// NOTE: this is a little weird, but props.show is _not_ reactive, so we cannot watch it
+//   directly and wrapping it in toRef would be misleading, since its readonly, so
+//   computed is a fair compromise
+const show = computed(() => props.show);
+
+watch(show, (newValue) => {
+  if (newValue) {
+    if (!props.edit) {
+      stepStore.resetActiveStep();
+    }
+  }
+});
+
+const onCreateOrUpdate = async () => {
+  emit("close");
+  if (props.edit) {
+    await stepStore.updateStep();
+  } else {
+    await stepStore.createStep();
+  }
+};
 
-const onCreate = async () => {
-  showAddStepModal.value = false;
-  await stepStore.createStep();
+const onClose = async () => {
+  emit("close");
 };
 </script>
 
 <template>
-  <v-dialog
-    activator="parent"
-    v-model="showAddStepModal"
-    persistent
-    v-if="trackStore.current && stepStore.newStepValues && stepStore.newStepEnabled"
-  >
+  <v-dialog v-model="show" persistent>
     <v-card class="pa-2">
-      <v-card-title class="text-h5">{{ $t("entity.step.add") }}</v-card-title>
-      <v-container>
+      <v-card-title class="text-h5">{{ edit ? $t("entity.step.edit") : $t("entity.step.add") }}</v-card-title>
+      <!-- TODO v-else would be nice-->
```

Comment:
Does it make sense to have this as a GitHub issue or are inline `todo`s the more preferred way?

Reply:
For this one, we can keep the inline TODO. The project is not in a state where I would go over it with this fine granularity. Going back over my code, I felt like the data modelling has some problems, but more than that the whole vue ui coding approach that I adopted here feels wrong to me. 
I want to take some Mondays and explore other frameworks, like Svelte or Leptos, and see if these feels better for this type of project.